api/v1/testkube.yaml

```diff
@@ -649,6 +718,129 @@ paths:
 
 components:
   schemas:
+
+    ObjectRef:
+      type: object
+      properties:
+        namespace:
+          type: string
+        name:
+          type: string
+
+
+    Test:
+      type: object
+      properties:
+        before:
+          $ref: "#/components/schemas/TestStep"
+          description: Run this step before whole suite
+        steps:
+          type: array
+          items:
+            $ref: "#/components/schemas/TestStep"
+          description: Steps to run
+        after:
+          $ref: "#/components/schemas/TestStep"
+          description: Run this step after whole suite
+        repeats:
+          type: integer
+          default: 1
+
+    TestStep: 
+      type: object 
+      required:
+        - name
+      properties:
+        name: 
+          type: string
+          description: step name (in case of script it'll be script name)
+
+    TestStepExecuteScript: 
+      allOf:     # Combines the BasicErrorModel and the inline model
+        - $ref: '#/components/schemas/TestStep'
+        - type: object
+          required:
+            - namepace 
+            - name
+          properties:
+            namespace:
+              type: string
+            name:
+              type: string
+            stopOnFailure:
+              type: boolean
+              default: true
+
+    TestStepDelay: 
+      allOf:     # Combines the TestStep and the inline model
+        - $ref: '#/components/schemas/TestStep'
+        - type: object
+          required:
+            - duration 
+          properties:
+            delay:
+              type: integer
+              default: 0 
+              description: delay in milliseconds
+        
+          
+    TestExecution:
+      type: object
+      description: API server test scripts executions container
+      properties:
+        id:
+          type: string
+          description: execution id
+          format: bson objectId
+        name:
+          type: string
+          description: "execution name"
+        envs:
+          type: object
+          description: "environment variables passed to executor"
+          additionalProperties:
+            type: string
+          example:
+            record: "true"
+            prefix: "some-"
+        params:
+          type: object
+          description: "execution params passed to executor converted to vars for usage in tests"
+          additionalProperties:
+            type: string
+          example:
+            users: "3"
+            prefix: "some-"
+        startTime:
+          type: string
+          description: 'test start time'
+          format: date-time
+        endTime:
+          type: string
+          description: 'test end time'
+          format: date-time
+        executionsResults:
```

Comment:
stepExecutionResults?  - all Steps should have a StepExecutionResult, some simpler than others?

Reply:
maybe stepResult - to not make those fields too hard to read for user ? @olensmar